File: 20070224/Lua/solve.lua
require("cycle_length")

while 1 do
	local line = io.read()
	local _, _, i, j = string.find(line, "(%d+) (%d+)")
	if i == nil or j == nil then
		error("ERROR.")
	end
	if i > j then
		local temp = i
		i = j
		j = i
	end
	local maxlength = 0
	for n = i, j do
		length = CycleLength:getLength(n)
		if length > maxlength then
			maxlength = length
		end
	end
	io.write(i, " ", j, " ", maxlength, "\n")
end

File: 20070224/Ruby/solve.rb
require 'CycleLength'

cyclelength = CycleLength.new

while 1
	i, j = gets.split(" ")
	i, j = i.to_i, j.to_i
	i, j = j, i if i > j
	maxlength = 0
	for n in (i..j)
		length = cyclelength.getLength(n)
		maxlength = length if length > maxlength
	end
	puts i.to_s + " " + j.to_s + " " + maxlength.to_s
end
